test/doctool/test-doctool-html.js

```diff
@@ -126,6 +121,10 @@ const testData = [
     'href="#foo_see_also" id="foo_see_also">#</a></span></h2><p>Check' +
     'out also<a href="https://nodejs.org/">this guide</a></p>'
   },
+  {
+    file: fixtures.path('document_with_special_heading.md'),
+    html: '<title>Sample markdown with special heading |',
```

Comment:
For the test to pass, the content in the `<body>` of the document must be present.

```suggestion
    html:
      '<h1>Sample <code>markdown</code> with <em>special</em>' +
      '<strong>heading</strong><span><aclass="mark"href="#foo_sample_' +
      'markdown_with_special_heading"id="foo_sample_markdown_with_special_' +
      'heading">#</a></span></h1>',
```

Reply:
No, you're missing entirely the point of the test! I want to test if the `<title>` contains the full text, the HTML content is completely irrelevant for this test.

Reply:
@aduh95, this is the expected output (if you remove the prose below the H1), not the title.

That's how I know this test was never run. I don't see how I can be wrong about that.

Run it yourself, you'll see.

```bash
node test/doctool/test-doctool-html.js
```

Reply:
> I want to test if the `<title>` contains the full heading text, the HTML content is completely irrelevant for this test.

The `<h1>` content is what eventually becomes the `<title>`, so this is the test for that.

Reply:
Nope, if you go check out the HTML content of https://nodejs.org/docs/latest/api/module.html, you can see that:

* `<h1>` contains the full expected text: `<h1>Modules: <code>module</code> API<span><a …`
* `<title>` does not: `<title>Modules:  | …`

Reply:
> That's how I know this test was never run. I don't see how I can be wrong about that.

Are you calling me a liar? Did you run the test?

Reply:
Okay, so the way that this test file is set up is probably going to need to change then because as of right now it seems to me like what it is looking for in `EXPECTED` is the content within the `<body>`, which isn't what we really want here.

There's a comment about not having an HTML parser, so we might need to make some adjustments to accommodate.

Reply:
> it seems to me like what it is looking for in EXPECTED is the content within the <body>

Not really, it's checking if the  `EXPECTED` value is contained within the HTML output – so `<html>`, not just `<body>`. That's very convenient to me because this PR does not touch the `<body>` content but the `<head>` content.

Other tests in this file focus on the `<body>` content, and the one I'm adding is focused on the `<title>` element. If you ask me, this test is doing a great job as is.

If you want to propose changes to the current test, I think it should be done in another PR.

Reply:
Ah, I see why it wasn't working for me now &mdash; my mistake.